cl/corpus_importer/management/commands/harvard_merge.py

```diff
@@ -0,0 +1,613 @@
+    :return: None
+    """
+    cluster = OpinionCluster.objects.get(id=cluster_id)
+    harvard_date_filed = harvard_data.get("decision_date")
+    cluster_date_filed = cluster.date_filed
+    merge_cluster_dates(
+        cluster_id, "date_filed", (harvard_date_filed, cluster_date_filed)
+    )
+
+
+def merge_overlapping_data(cluster_id: int, clean_dictionary: dict) -> None:
+    """Merge overlapping data
+
+    :param cluster_id: the cluster id
+    :param clean_dictionary: the dictionary of data to merge
+    :return: None
+    """
+    if clean_dictionary != {}:
+        logging.info(f"Merging complete for: {cluster_id}")
+        return
+
+    long_fields = [
+        "syllabus",
+        "summary",
+        "history",
+        "headnotes",
+        "correction",
+        "cross_reference",
+        "disposition",
+    ]
+
+    for field_name in clean_dictionary.keys():
+        if field_name in long_fields:
+            merge_long_fields(
+                cluster_id,
+                field_name,
+                clean_dictionary.get(field_name),
+            )
+        elif field_name in ["other_dates"]:
+            merge_cluster_dates(
+                cluster_id,
+                field_name,
+                clean_dictionary.get(field_name),
+            )
+        elif field_name == "judges":
+            merge_judges(
+                cluster_id,
+                clean_dictionary.get(field_name),
+            )
+        elif field_name == "attorneys":
+            merge_strings(
+                cluster_id,
+                field_name,
+                clean_dictionary.get(field_name, ""),
+            )
+        else:
+            logging.info(f"Field not considered in the process: {field_name}")
+
+
+def update_docket_source(cluster_id: int) -> None:
+    """Update docket source and complete
+
+    :param cluster_id: the cluster id
+    :return: None
+    """
+    docket = OpinionCluster.objects.get(id=cluster_id).docket
+    source = docket.source
+    docket.source = Docket.HARVARD + source
+    docket.save()
+
+
+def update_cluster_source(cluster_id: int) -> None:
+    """Update cluster source
+
+    :param cluster_id: cluster id to update
+    :return: None
+    """
+    cluster = OpinionCluster.objects.get(id=cluster_id)
+    if cluster.source == "C":
+        # court website source updated to court website merged with Harvard
+        cluster.source = "CU"
+        cluster.save()
+
+
+def merge_opinion_clusters(cluster_id: int, fastcase: bool = False) -> None:
+    """Merge opinion cluster, docket and opinion data from Harvard
+
+    :param cluster_id: The cluster ID to merger
+    :return: None
+    """
+    harvard_data = read_json(cluster_id)
+    if harvard_data:
+        source = get_data_source(harvard_data)
+        try:
+            with transaction.atomic():
+                if source == "Fastcase" or fastcase:
+                    # Do something special for fastcase
+                    pass
```

Comment:
? this doesnt look like it is ready.

Reply:
I don't know what extra code you need for fastcase data, that's why i left it empty,  just added the flag for the command

Reply:
what do you expect the fastcase flag to do? I was also wondering if fastcase should be listed as a source for OpinionCluster or Docket